### PR TITLE
support slash in `name` and `label` property of `azurerm_app_configuration_feature`

### LINKED ISSUE #19437: azurerm_app_configuration_feature breaks state when using slashes
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.5

### AzureRM Provider Version

3.33.0

### Affected Resource(s)/Data Source(s)

azurerm_app_configuration_feature

### Terraform Configuration Files

```hcl
data "azurerm_client_config" "current" {}

resource "azurerm_role_assignment" "appconf_dataowner" {
  scope                = var.resource_group_id
  role_definition_name = "App Configuration Data Owner"
  principal_id         = data.azurerm_client_config.current.object_id
}

resource "azurerm_app_configuration" "appconfig" {
  name                = "appconf-fred"
  resource_group_name = var.resource_group_name
  location            = var.resource_group_location
  sku                 = "standard"
  tags                = var.tags
}

resource "azurerm_app_configuration_feature" "feature_toggle" {
  configuration_store_id = azurerm_app_configuration.appconfig.id
  name                   = "my/test"
  description            = "My feature"
  enabled                = true
}
```

### Debug Output/Panic Output

```shell
module.appconfiguration[0].azurerm_app_configuration_feature.feature_toggle: Creating...
╷
│ Error: while parsing resource ID: while parsing resource ID: The number of path segments is not divisible by 2 in "subscriptions/XXXX/resourceGroups/rg-terraform-fred/providers/Microsoft.AppConfiguration/configurationStores/appconf-fred/AppConfigurationFeature/my/test/Label/\x00"
│
│   with module.appconfiguration[0].azurerm_app_configuration_feature.feature_toggle,
│   on appconfig\main.tf line 20, in resource "azurerm_app_configuration_feature" "feature_toggle":
│   20: resource "azurerm_app_configuration_feature" "feature_toggle" {
│
│ while parsing resource ID: while parsing resource ID: The number of path segments is not divisible by 2 in
│ "subscriptions/XXXX/resourceGroups/rg-terraform-fred/providers/Microsoft.AppConfiguration/configurationStores/appconf-fred/AppConfigurationFeature/my/test/Label/\x00"
╵
```

### Expected Behaviour

A feature succesfully being added in app configuration with name 'my/feature' and with key '.appconfig.featureflag/my/test'.

### Actual Behaviour

The feature is added, but terraform is failing with an error after creation. State is also 'corrupted', meaning I cannot fix it by changing my feature configuration. (for example by changing the slash into a dash)

### Steps to Reproduce

1. setup app configuration as described in the documentation: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/app_configuration_feature
2. add a slash in the name
3. execute terraform plan -> no errors
4. execute terraform apply -> fails with an error (feature is created in azure app configuration)
5. remove the slash from the name
6. execute terraforn plan -> fails with the same error

### Important Factoids

_No response_

### References

A fix was done here https://github.com/hashicorp/terraform-provider-azurerm/issues/16835 but the focus was on the labels if I'm not mistaken?
A fix for supporting the slash in keys was done here: https://github.com/hashicorp/terraform-provider-azurerm/issues/13852

Both fixed were done for azurerm_app_configuration_key but should also be executed for azurerm_app_configuration_feature?

### PR DESCRIPTION
**Pull request by ziyeqf**
fix #19437

Screenshot 2022-11-29 at 15 38 50

### CODE REVIEW

File: internal/services/appconfiguration/app_configuration_feature_resource.go [hunk 1]
```diff
@@ -42,6 +44,15 @@ type FeatureResourceModel struct {
 	TargetingFilters     []TargetingFilterAudience    `tfschema:"targeting_filter"`
 }
 
+func (FeatureResource) StateUpgraders() sdk.StateUpgradeData {
```
Comment (katbyte) [inline internal/services/appconfiguration/app_configuration_feature_resource.go:47]: could you explain why we need a state migration here? no existing resources should have a / ?

File: internal/services/appconfiguration/app_configuration_feature_resource.go [hunk 2]
```diff
@@ -175,8 +186,8 @@ func (k FeatureResource) Create() sdk.ResourceFunc {
 
 			appCfgFeatureResourceID := parse.AppConfigurationFeatureId{
 				ConfigurationStoreId: model.ConfigurationStoreId,
-				Name:                 model.Name,
-				Label:                model.Label,
+				Name:                 url.QueryEscape(model.Name),
+				Label:                url.QueryEscape(model.Label),
```
Comment (katbyte) [inline internal/services/appconfiguration/app_configuration_feature_resource.go:190]: the API don't accept the value if its not escaped? the error was in parsing?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/appconfiguration/app_configuration_feature_resource.go (hunk 1 above)
Comment (ziyeqf) [reply internal/services/appconfiguration/app_configuration_feature_resource.go:47]: there is a possibility existing resources have "/" in its id, and the state file will be "broken" like the issue described #19437. I think we add a state migration here will help

File: internal/services/appconfiguration/app_configuration_feature_resource.go (hunk 2 above)
Comment (ziyeqf) [reply internal/services/appconfiguration/app_configuration_feature_resource.go:190]: the API does accept unescaped value, escaping here is to work for `appCfgFeatureResourceID.ID()`.  Do you mean we only need to patch the parsing process?

**Comment by ziyeqf** [comment]
running testsing

File: internal/services/appconfiguration/app_configuration_feature_resource.go (hunk 2 above)
Comment (ziyeqf) [reply internal/services/appconfiguration/app_configuration_feature_resource.go]: ok, modified. I was worried using a regexp in parsing process maybe not stable as we except so I used escape to do this patch.

File: internal/services/appconfiguration/app_configuration_feature_resource.go (hunk 1 above)
Comment (ziyeqf) [reply internal/services/appconfiguration/app_configuration_feature_resource.go]: removed in new commit

**Review by katbyte — APPROVED** [review]
Thanks LGTM 🍄

**Pull request merged**